src/widgets/modals/SearchForm.js

```diff
@@ -0,0 +1,159 @@
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2020
+ */
+
+import React, { useMemo, useState } from 'react';
+import { View, StyleSheet, FlatList, Text } from 'react-native';
+import PropTypes from 'prop-types';
+
+import { FormControl } from '../FormControl';
+
+import { getFormInputConfig } from '../../utilities/formInputConfigs';
+
+import { APP_FONT_FAMILY, DARK_GREY } from '../../globalStyles';
+import { TouchableOpacity } from 'react-native-gesture-handler';
+import { queryPatientApi, queryPrescriberApi } from '../../utilities/network/lookupApi';
+
+const RECORD_TYPES = {
+    PATIENT: 'patient',
+    PRESCRIBER: 'prescriber',
+}
+
+const FORM_CONFIGS = {
+    [RECORD_TYPES.PATIENT]: 'searchPatient',
+    [RECORD_TYPES.PRESCRIBER]: 'searchPrescriber',
+}
+
+const LIST_CONFIGS = {
+    [RECORD_TYPES.PATIENT]: [ 'firstName', 'lastName', 'dateOfBirth' ],
+    [RECORD_TYPES.PRESCRIBER]: [ 'firstName', 'lastName', 'registrationCode' ],
+}
+
+const SearchListItemColumn = ({ value, type }) => (
+    <View style={localStyles.columnContainer}>
+        <Text style={localStyles.text}>
+            {type === 'date' ? value.toDateString() : value}
+        </Text>
+    </View>
+);
+
+const SearchListItem = ({ listItem, listConfig }) => {
+    const onPress = item => console.log(item);
+    const columns = listConfig.map(({ key, type}) => {
+        const value = listItem[key];
+        return <SearchListItemColumn key={key} value={value} type={type}/>;
+    });
+    return (
+        <TouchableOpacity onPress={onPress} style={localStyles.rowContainer}>
+            {columns}
+        </TouchableOpacity>
+    );
+}
+
+export const SearchForm = ({dataSource}) => {
+    const [data, setData] = useState([]);
+
+    if (!RECORD_TYPES.includes(dataSource)) return null;
+
+    const configKey = useMemo(() => FORM_CONFIGS[dataSource], [dataSource]);
+    const formConfig = useMemo(() => getFormInputConfig(configKey), [configKey]);
+
+    const listConfig = useMemo(() => {
+        const keys = LIST_CONFIGS[dataSource];
+        const keyTypes = keys.reduce((acc, key) => ({ ...acc, [key]: formConfig.find(config => config.key === key)?.type }), {});
+        return keys.map(key => ({ key, type: keyTypes[key] }));
+    }, [formConfig]);
```

Comment:
this is all needed because of the date? I think it might be nicer to have the response have the details formatted prior to getting to this component, if that makes sense? There is already complexity I imagine where the request is being managed, so I think putting it there?

Reply:
At the moment just the date, yeah, but would also be needed for e.g. numbers, etc. My thinking was that its not so much the formatting of the response, it's more how to display it in a list. As more info will be being returned than is displayed, it seemed a good idea to separate the view from the data, and leave how to display the data to the component. Also a bit cleaner as if the row is selected, the data doesn't have to be converted back again to enter the record into the database.

Not sure if that's a good argument or not :man_shrugging:. I'm cool with making the change if you still think its better!